content/blog/integrating-istio-and-spire.md

```diff
@@ -0,0 +1,463 @@
+
+NAME                                       DESIRED   CURRENT   READY   AGE
+replicaset.apps/details-v1-f8957ccb4       1         1         1       37d
+replicaset.apps/productpage-v1-cfb4bc854   1         1         1       37d
+replicaset.apps/ratings-v1-65cd6fbcd8      1         1         1       37d
+replicaset.apps/reviews-v1-55f769fb78      1         1         1       37d
+replicaset.apps/reviews-v2-6b7c798cc8      1         1         1       37d
+replicaset.apps/reviews-v3-695c7f59db      1         1         1       37d
+```
+
+Once everything is up, all workloads would get registered under SPIRE server.
+
+**4.6** You can verify the registration of workloads using the following command:
+
+```shellsession
+kubectl exec <spire-server_pod_name> -n spire -c spire-server -- ./bin/spire-server entry show
+```
+
+Verify that every workload with same label as clusterSPIFFEID CRD’s match label is registered in the server.
+
+![](/img/server-entries.png)
+
+**4.7** Verify that the certificate issuer of workloads is SPIRE using following commands for each workload.
+
+```shellsession
+istioctl proxy-config secret <pod_name> -n <namespace_name> -o json | jq -r '.dynamicActiveSecrets[0].secret.tlsCertificate.certificateChain.inlineBytes' | base64 --decode > chain.pem
+```
+
+```shellsession
+k8s-spiffe-integ-master-7j7fh-m67q9:~ openssl x509 -in chain.pem -text | grep SPIRE
+ Subject: C = US, O = SPIRE, x500UniqueIdentifier = e2f9c35b9198e1824373e874b13287d0
+```
+
+You should also check the same for ingress-gateway pod in Istio-system namespace and verify that your deployed workloads and ingress-gateway has the same issuer.
+
+#### Step 5: Open the application outside traffic
+
+The Bookinfo application is deployed but not accessible from the outside. To make it accessible, you need to create an Istio Ingress Gateway, which maps a path to a route at the edge of your mesh.
+
+**5.1** Associate this application with the Istio gateway:
+
+```shellsession
+kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml -n bookinfo
+```
+
+**5.2** Ensure that there are no issues with the configuration:
+
+```
+k8s-spiffe-integ-master-7j7fh-m67q9:~ # istioctl analyze -n bookinfo
+
+✔ No validation issues found when analyzing namespace: bookinfo.
+```
+
+**5.3** Execute the following command to determine if your Kubernetes cluster is running in an environment that supports external load balancers:
+
+```shellsession
+k8s-spiffe-integ-master-7j7fh-m67q9:~ kubectl get svc istio-ingressgateway -n istio-system
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                      AGE
+istio-ingressgateway   LoadBalancer   10.105.191.32   172.16.17.5   15021:30189/TCP,80:30392/TCP,443:30566/TCP   32d
+```
+
+If the EXTERNAL-IP value is set, your environment has an external load balancer; if not, then set the external load balancer first then follow further steps.
```

Comment:
Recommend breaking this out into two sentences:
If the EXTERNAL-IP value is set, your environment has an external load balancer. If not, then set the external load balancer first before following the next steps.